### ISSUE TITLE
[Bug] Sometimes when switching tabs via Ctrl-Tab, the old tab remains highlighted

### ISSUE THREAD

**Issue by Ennea** [piroor/treestyletab#3488]
## Abstract

Sometimes (maybe 1 out of 20 cases) when switching tabs by just tapping Ctrl-Tab (tab cycling is set to most recently used order), the previous tab remains highlighted, as if it had been selected with a Ctrl-Click. This started happening with v4.0.

Edit: Just after writing this ticket, this bug also happened when I was switching tabs with Ctrl-PgUp/Ctrl-PgDown.

Edit2: I've observed this a little more: the bug is strictly visual. The "selection" itself is still just a singular tab, and the context menu also reflects that.

## Steps to reproduce

 1. Keep tapping Ctrl-Tab to switch between the two most recently open tabs
 2. Observe that sometimes, the last tab will remain highlighted after switching away from it

## Expected result

Last tab should not remain highlighted.

## Actual result

Last tab remains highlighted.

## Environment

 * Platform (OS): Linux
 * Version of Firefox: 124.0b9
 * Version (or revision) of Tree Style Tab: 4.0.3

**Comment by piroor** [comment]
I couldn't reproduce this with TST 4.0.3 + Firefox 124.0b9 + Ubuntu 22.04.

> Edit2: I've observed this a little more: the bug is strictly visual. The "selection" itself is still just a singular tab, and the context menu also reflects that.

It may be a problem of the hardware acceleration of Firefox. Could you try with disabled HA?

**Comment by Ennea** [comment]
I've checked the markup of the sidebar (via `about:debugging`) and the tab in question had the `highlighted` class, so I suspect it's not going to be related to the hardware acceleration. Perhaps it has something to do with my large number of tabs, too.

I will try to reproduce this on a fresh profile of Firefox and get back to you once I've figured out what constellation of things lead to it.

**Comment by Ennea** [comment]
Okay, I've got a repro. I've also realized that this is only visible to me when I select the "system theme" in the themes. The dark and light themes don't appear to have the little blue bar in the `highlighter` element, but the system one does, apparently. The classes are of course still "broken" in the other themes, it's just not visible.

To reproduce, the number of open tabs does not seem to matter. Follow these steps:
1. Click on a tab in TST
2. Move to the tab above or below using Ctrl-PgUp/Ctrl-PgDown
3. Move back to the original tab clicked on in step 1, again via Ctrl-PgUp/Ctrl-PgDown

After that, the tab clicked on in step 1 is the active tab, but the tab moved to in step 2 still has the `highlighted` class on its element. Reproduced on a fresh profile.
